### ISSUE TITLE
completions/git: diff & difftool need adjustments for deleted files

### ISSUE THREAD

**Issue by z3ntu** [fish-shell/fish-shell#8090]
As done for git reset ("# Deleted files seem to need a "--" separator."), you also need two dashes for diff-ing deleted files

```
mkdir /tmp/test
cd /tmp/test
git init
touch test
git add test
git commit -m "foo"
rm test
git diff <TAB> test
```
```
fatal: ambiguous argument 'test': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```
(both for diff and difftool)

**Comment by lgarron** [comment]
I still have this issue in `fish` 3.1.1. Is it possible that `fish` might not get this fix while updating?

**Comment by lgarron** [comment]
> I still have this issue in `fish` 3.1.1. Is it possible that `fish` might not get this fix while updating?

Hmm, this seems to be due to my workarounds for https://github.com/fish-shell/fish-shell/issues/3342 (which I now use for a different huge repo).

Previous: https://github.com/lgarron/dotfiles/blob/5eb861c928ed3e8e6b187ce838b64bcd37581edd/dotfiles/fish-github/.config/fish/completions/git.fish
Latest: https://github.com/lgarron/dotfiles/blob/1baad4798f7806a924b874ade2ffa222c73c7352/dotfiles/fish-github/.config/fish/completions/git.fish

I [hopefully no longer need these](https://github.blog/2021-08-11-githubs-engineering-team-moved-codespaces/), but I'd appreciate updated tips on how to stub out super slow `git` completions in the latest `fish` version.

**Comment by krobelus** [comment]
> I still have this issue in fish 3.1.1

@lgarron the fix was included in 3.3.1

it is unrelated to your workarounds, which seem fine, except that `^ /dev/null` should be `2>/dev/null`

**Comment by lgarron** [comment]
> @lgarron the fix was included in 3.3.1

Sorry, I meant 3.3.1. 🤦‍♀️

> it is unrelated to your workarounds, which seem fine, except that `^ /dev/null` should be `2>/dev/null`

Aha! Thanks. :-D
